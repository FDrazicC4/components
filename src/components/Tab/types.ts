import { StyledParentComponent } from '../types';

export interface Props extends StyledParentComponent {

    /**
     * If true, the tab will be disabled.
     */
    disabled?: boolean,

    /**
     * The icon element.
     */
    icon?: string,

    /**
     * The label for the tab
     */
    label?: string,

    /**
     * You can provide your own value.  Otherwise, we fall back to the child position index.
     */
    value?: any,

    /**
     * The tab link, if you'd like to link somewhere instead of just showing or hiding elements
     */
    href?: string,
};
