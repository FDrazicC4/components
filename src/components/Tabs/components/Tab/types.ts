import React, {Node, ReactNode} from "react";

export type Variants = 'standard' | 'icon';
export type Colors = 'primary' | 'secondary' | 'success' | 'danger' | 'warning' | 'info' | 'light' | 'dark' | 'link';
export type Sizes = 'small' | 'medium' | 'large';


export interface Props {
    /**
     * The content inside the tabs to be rendered or the icon to render if `variant="icon"`. Usually just a string, but can be any valid JSX.
     */
    children?: ReactNode,
    /**
     * Fires when button is clicked.
     */
    onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void,
    /**
     * Classes to be passed through to the component.
     */
    className?: string,
    /**
     * If set to "standard", the content of the button will be rendered as is. If set to "icon", the content is expected
     * to be a string and will determine what icon to render.
     */
    highlightStyle?: 'underline' | 'background',
    inline?: boolean,
    noSpread?: boolean,
    shadowless?: boolean,
    shouldPreserveContent?: boolean,
}
