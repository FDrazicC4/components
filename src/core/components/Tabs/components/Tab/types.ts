import {ReactNode} from "react";

export interface Props {
    isActive?: boolean,
    className?: string,
    children?: ReactNode,
};
