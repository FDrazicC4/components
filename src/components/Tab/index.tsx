import React from 'react';
import MuiTab from '@material-ui/core/Tab';
import Typography from '@components/Typography';
import Icon from '@components/Icon';
import { Props } from './types';

/**
 * The `<Tab />` component shows a tab within the `<TabList />` component.  
 * 
 * Tabs organize and allow navigation between groups of content or different URL's.
 */
const Tab = React.forwardRef(({ label = '', icon, ...otherProps }: Props, ref: React.Ref<any>): React.ReactElement => {
    const classes = {
        root: 'brew-Tab',
        labelIcon: 'brew-Tab--labelIcon',
        textColorInherit: 'brew-Tab--textColorInherit',
        textColorPrimary: 'brew-Tab--textColorPrimary',
        textColorSecondary: 'brew-Tab--textColorSecondary',
        selected: 'brew-Tab--selected',
        disabled: 'brew-Tab--disabled',
        fullWidth: 'brew-Tab--fullWidth',
        wrapped: 'brew-Tab--wrapped',
        wrapper: 'brew-Tab__wrapper',
    };

    return (
        <MuiTab
            {...otherProps}
            classes={classes}
            ref={ref}
            icon={icon ? <Icon className="brew-Tab__icon">{icon}</Icon> : undefined}
            label={<Typography className="brew-Tab__label">{label}</Typography>}
        />
    );
});

export default Tab;
