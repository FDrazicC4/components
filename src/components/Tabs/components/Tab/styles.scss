@import '../../../../styles/utilities';

.tab {
	color: color(grey, 500);
	padding: 0.75rem 1rem;
}

.isActive {
	color: color(brand, primary);

	&:after {
		position: absolute;
		bottom: 0;
		left: 0;
		width: 100%;
		height: 4px;
		background-color: color(brand, primary);
	}
}
