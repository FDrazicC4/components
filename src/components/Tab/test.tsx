import React from 'react';
import renderer from 'react-test-renderer';
import Tab from '.';


describe('Tab', () => {

    it('Renders correctly with defaults', () => {
        const tree = renderer
            .create(<Tab />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Renders correctly with property variants', () => {
        const tree = renderer
            .create((
                <React.Fragment>
                    <Tab label="label 1" />
                    <Tab label="label 2" />
                    <Tab icon="add_circle" label="label 1" />
                    <Tab icon="add_circle" label="label 1" />
                    <Tab icon="add_circle" label="label 1" disabled />
                </React.Fragment>
            ))
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

});
