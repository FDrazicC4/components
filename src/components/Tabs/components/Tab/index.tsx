import React from 'react';
import clsx from 'clsx';
import Typography from '../../../Typography';
import { Props } from './types';
import styles from './styles.scss';


/**
 * The `Tab` component represents a clickable button, which can be used in forms or anywhere in the application
 * that needs simple, standard button functionality.
 */
function Tab({
    children,
    className,
    ...otherProps
}: Props) {


    const tabClasses = clsx(
        styles.wrapper,
        className,
    );


    return (
        <span className={tabClasses}>
            <Typography>{children}</Typography>
        </span>
    );


}


export default Tab;