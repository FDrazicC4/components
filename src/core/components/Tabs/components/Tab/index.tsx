import React from 'react';
import clsx from "clsx";
import styles from './styles.scss';
import { Props } from './types';
import Typography from "../../../Typography";

function Tab({ isActive, className, children, ...otherProps }: Props) {
    const classes = clsx(
        styles.tab,
        { [styles.isActive]: !!isActive },
        className,
    );
    return (
        <div className={classes} {...otherProps}><Typography>{children}</Typography></div>
    )
}

export default Tab;
