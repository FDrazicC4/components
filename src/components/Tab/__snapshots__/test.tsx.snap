// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab Renders correctly with defaults 1`] = `
<button
  className="MuiButtonBase-root MuiTab-root brew-Tab MuiTab-textColorInherit brew-Tab--textColorInherit"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="tab"
  tabIndex={0}
  type="button"
>
  <span
    className="MuiTab-wrapper brew-Tab__wrapper"
  >
    <p
      className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-body1 brew-Typography--align-inherit brew-Tab__label MuiTypography-body1"
    >
      
    </p>
  </span>
</button>
`;

exports[`Tab Renders correctly with property variants 1`] = `
Array [
  <button
    className="MuiButtonBase-root MuiTab-root brew-Tab MuiTab-textColorInherit brew-Tab--textColorInherit"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="tab"
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiTab-wrapper brew-Tab__wrapper"
    >
      <p
        className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-body1 brew-Typography--align-inherit brew-Tab__label MuiTypography-body1"
      >
        label 1
      </p>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>,
  <button
    className="MuiButtonBase-root MuiTab-root brew-Tab MuiTab-textColorInherit brew-Tab--textColorInherit"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="tab"
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiTab-wrapper brew-Tab__wrapper"
    >
      <p
        className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-body1 brew-Typography--align-inherit brew-Tab__label MuiTypography-body1"
      >
        label 2
      </p>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>,
  <button
    className="MuiButtonBase-root MuiTab-root brew-Tab MuiTab-textColorInherit brew-Tab--textColorInherit MuiTab-labelIcon brew-Tab--labelIcon"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="tab"
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiTab-wrapper brew-Tab__wrapper"
    >
      <i
        aria-hidden={true}
        className="material-icons MuiIcon-root brew-Icon brew-Icon--color-inherit brew-Icon--size-inherit brew-Tab__icon MuiIcon-fontSizeInherit"
      >
        add_circle
      </i>
      <p
        className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-body1 brew-Typography--align-inherit brew-Tab__label MuiTypography-body1"
      >
        label 1
      </p>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>,
  <button
    className="MuiButtonBase-root MuiTab-root brew-Tab MuiTab-textColorInherit brew-Tab--textColorInherit MuiTab-labelIcon brew-Tab--labelIcon"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="tab"
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiTab-wrapper brew-Tab__wrapper"
    >
      <i
        aria-hidden={true}
        className="material-icons MuiIcon-root brew-Icon brew-Icon--color-inherit brew-Icon--size-inherit brew-Tab__icon MuiIcon-fontSizeInherit"
      >
        add_circle
      </i>
      <p
        className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-body1 brew-Typography--align-inherit brew-Tab__label MuiTypography-body1"
      >
        label 1
      </p>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>,
  <button
    className="MuiButtonBase-root MuiTab-root brew-Tab MuiTab-textColorInherit brew-Tab--textColorInherit Mui-disabled brew-Tab--disabled MuiTab-labelIcon brew-Tab--labelIcon Mui-disabled"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="tab"
    tabIndex={-1}
    type="button"
  >
    <span
      className="MuiTab-wrapper brew-Tab__wrapper"
    >
      <i
        aria-hidden={true}
        className="material-icons MuiIcon-root brew-Icon brew-Icon--color-inherit brew-Icon--size-inherit brew-Tab__icon MuiIcon-fontSizeInherit"
      >
        add_circle
      </i>
      <p
        className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-body1 brew-Typography--align-inherit brew-Tab__label MuiTypography-body1"
      >
        label 1
      </p>
    </span>
  </button>,
]
`;
